src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -484,22 +483,22 @@ public void addOnRankedMatchNotificationListener(Consumer<MatchmakerInfoMessage>
     fafService.addOnMessageListener(MatchmakerInfoMessage.class, listener);
   }
 
-  public CompletableFuture<Void> startSearchLadder1v1(Faction faction) {
+  public CompletableFuture<Void> startSearchLadder(RatingMode ratingMode, Faction faction) {
     if (isRunning()) {
-      log.debug("Game is running, ignoring 1v1 search request");
+      log.debug("Game is running, ignoring search request");
       return completedFuture(null);
     }
 
     if (!preferencesService.isGamePathValid()) {
       CompletableFuture<Path> gameDirectoryFuture = postGameDirectoryChooseEvent();
-      return gameDirectoryFuture.thenCompose(path -> startSearchLadder1v1(faction));
+      return gameDirectoryFuture.thenCompose(path -> startSearchLadder(ratingMode, faction));
     }
 
-    searching1v1.set(true);
+    searchingLadder.set(true);
 
-    return modService.getFeaturedMod(LADDER_1V1.getTechnicalName())
+    return modService.getFeaturedMod(ratingMode.getCorrespondingFeatureMod().getTechnicalName())
         .thenAccept(featuredModBean -> updateGameIfNecessary(featuredModBean, null, emptyMap(), emptySet()))
```

Comment:
Side note, wouldn't it be better to make sure the game files actually finished updating before we enter the queue? What if the queue pops and we are still downloading files?

Reply:
Huh? I have not touched the previous behavior that was done for a 1v1 queue. As I see we update the game first and only then (if an update is successful) we send a message to a server to start a search. The only thing which is done after the search is map downloading.

Reply:
Hmm I need to try it out again, but I thought the last time I did, it would enter me I to the queue before it finished updating.